### PR TITLE
[EP-2093] Last viewed place functionality

### PR DESCRIPTION
**Pull request by ancashoria**

### CODE REVIEW

File: packages/earth-map/src/components/list-item/component.tsx [hunk 1]
```diff
@@ -20,13 +23,24 @@ const ListItem = (props: IProps) => {
...
-    onClick = () => { },
+    list,
+    setPlacesSearch,
+    setIndexesSelected,
+    onClick,
   } = props;
 
+  // Default click action. Can be overritten by passing onClick prop
+  const onClickIndex = () => {
+    // @ts-ignore
```
Comment (corinacioloca) [inline packages/earth-map/src/components/list-item/component.tsx]: why ts-ignore?

File: packages/earth-map/src/components/sidebar/sidebar-layout-search/component.tsx [hunk 2]
```diff
@@ -20,11 +21,13 @@ const LayersDropdown: any = Keyframes.Spring({
 
 interface IProps {
   search?: any;
+  group?: string[];
   places?: any;
   layersPanel?: boolean;
   selected?: boolean;
   locationName?: string;
   locationOrganization?: string;
+  lastViewedPlace?: any;
```
Comment (corinacioloca) [inline packages/earth-map/src/components/sidebar/sidebar-layout-search/component.tsx]: IPlace or how the model is

File: packages/earth-map/src/components/sidebar/sidebar-layout-search/component.tsx [hunk 3]
```diff
@@ -84,21 +93,32 @@ const SidebarLayoutSearch = (props: IProps) => {
...
-const renderContent = (open, selected, showResults) => {
-  const onLocationPage = selected && open && showResults;
-  const onHomepage = !selected && showResults;
-
-  return (onLocationPage || onHomepage)
-    ? <PlacesResults />
-    : selected
-      ? <IndexSidebar />
-      : <FeaturedPlaces />
+const LastViewedPlace = ({ place }) => {
```
Comment (corinacioloca) [inline packages/earth-map/src/components/sidebar/sidebar-layout-search/component.tsx]: can we extract this in a separate component?

File: packages/earth-map/src/sagas/location/index.tsx [hunk 4]
```diff
@@ -93,10 +95,18 @@ function* toLocation({ payload, meta }) {
     yield put(setPlaceSelectedFilter(''));
     yield put(setPlaceSelectedSearch(''));
     yield put(setPlaceData(formattedData));
+    yield put(setLastViewedPlace({
+      id: data.id,
```
Comment (corinacioloca) [inline packages/earth-map/src/sagas/location/index.tsx:99]: destructure data

File: packages/earth-map/src/utils/store.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { Store } from 'redux';
+
+import { setPlacesSearch } from 'modules/places/actions';
+import { setUserGroup } from 'modules/user/actions';
+import { setMapStyle } from 'modules/map/actions';
+import { setLastViewedPlace } from 'modules/global/actions';
+
+/**
```
Comment (corinacioloca) [inline packages/earth-map/src/utils/store.ts]: why is this here? there's a folder store, utils should have small helper functions

File: packages/earth-map/src/utils/store.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        ...ephemeralState.sidebarState,
+        ...hasFilters && {
+          open: true
+        }
+      }));
+    }
+    if (ephemeralState.global && ephemeralState.global.lastViewedPlace) {
+      store.dispatch(setLastViewedPlace(ephemeralState.global.lastViewedPlace));
+    }
+    ephemeralState.user && store.dispatch(setUserGroup(ephemeralState.user.group));
```
Comment (corinacioloca) [inline packages/earth-map/src/utils/store.ts]: does ephemeralState state have a model? it seems it uses a lot of properties that are not defined

File: packages/earth-map/src/sagas/location/index.tsx (hunk 4 above)
Comment (ancashoria) [reply packages/earth-map/src/sagas/location/index.tsx:99]: That won't make the code more readable and would pollute the function scope with unnecessary variable declarations.

**Review by corinacioloca — APPROVED** [review]
